src/components/create-vm-dialog/createVmDialog.jsx

```diff
@@ -748,20 +752,66 @@ const UsersConfigurationRow = ({
     );
 };
 
+const SshKeysRow = ({
+    id, item, onChange, idx, removeitem,
+}) => {
+    const [finalized, setFinalized] = useState(false);
+
+    const validateKey = (key) => {
+        cockpit.script(`echo ${key} | ssh-keygen -l -f /dev/stdin`, {})
+                .then(() => setFinalized(true))
+                .catch(() => setFinalized(false));
+    };
+
+    return (
+        <Grid id={id}>
+            <GridItem span={11}>
+                {finalized
+                    ? <span>{item.value}</span>
+                    : <TextArea id="ssh-key"
+                        value={item.value || ""}
+                        onChange={(_, value) => {
+                            onChange(idx, "value", value);
+                            validateKey(value);
+                        }}
+                        rows="3" />
+                }
+            </GridItem>
+            <GridItem span={1} className="pf-m-1-col-on-md remove-button-group">
+                <Button variant='plain'
+                    className="btn-close"
+                    id={id + "-btn-close"}
+                    isSmall
+                    aria-label={_("Remove item")}
+                    icon={<TrashIcon />}
+                    onClick={() => removeitem(idx)} />
+            </GridItem>
+        </Grid>
```

Comment:
These 11 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1150-20230911-122710-2dcbda42-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)